src/AGTHARN/FlyPE/Main.php

```diff
@@ -27,319 +27,177 @@
 
 namespace AGTHARN\FlyPE;
 
-use pocketmine\entity\Entity;
-use pocketmine\command\Command;
-use pocketmine\command\CommandSender;
-use pocketmine\event\entity\EntityDamageByEntityEvent;
-use pocketmine\event\entity\EntityLevelChangeEvent;
-use pocketmine\event\player\PlayerItemConsumeEvent;
-use pocketmine\event\inventory\InventoryPickupItemEvent;
-use pocketmine\event\block\BlockBreakEvent;
-use pocketmine\event\block\BlockPlaceEvent;
-use pocketmine\event\player\PlayerJoinEvent;
-use pocketmine\event\player\PlayerDropItemEvent;
-use pocketmine\utils\Config;
+use pocketmine\plugin\PluginBase;
 use pocketmine\utils\TextFormat as C;
-use pocketmine\event\Listener;
 use pocketmine\Player;
-use pocketmine\item\Item;
-use pocketmine\plugin\PluginBase;
+
+use AGTHARN\FlyPE\commands\FlyCommand;
+use AGTHARN\FlyPE\tasks\ParticleTask;
+use AGTHARN\FlyPE\lists\ParticleList;
+
 use jojoe77777\FormAPI\SimpleForm;
 use JackMD\UpdateNotifier\UpdateNotifier;
 use onebone\economyapi\EconomyAPI;
 
-class Main extends PluginBase implements Listener {
-
-    private $config;
+class Main extends PluginBase {
+	
+	/**
+	 * instance
+	 * 
+     * @var Main
+     */
+    public static $instance;
+	
+	/**
+	 * onEnable
+	 *
+	 * @return void
+	 */
+	public function onEnable(): void {
+		self::$instance = $this;
 
-    public function onEnable(){
-        $this->getServer()->getPluginManager()->registerEvents($this, $this);
+		$this->getServer()->getPluginManager()->registerEvents(new EventListener($this), $this);
+		$this->getServer()->getCommandMap()->register("fly", new FlyCommand("fly", $this));
+		if ($this->getConfig()->get("enable-fly-particles") === true) {
+			$this->getScheduler()->scheduleRepeatingTask(new ParticleTask($this), $this->getConfig()->get("fly-particle-rate"));
+		}
 
-        $this->saveResource('config.yml');
-        $this->config = new Config($this->getDataFolder() . 'config.yml', Config::YAML);
-	    
-        $configversion = $this->config->get("config-version");
-	    
-	    if($configversion < "2"){
+		if ($this->getConfig()->get("config-version") < "3") {
 		    $this->getLogger()->warning("Your config is outdated! Please delete your old config to get the latest features!");
 		    $this->getServer()->getPluginManager()->disablePlugin($this);
-		    //im sorry dylan
 	    }
 	    UpdateNotifier::checkUpdate($this->getDescription()->getName(), $this->getDescription()->getVersion());
-    }
-
-    public function onPlayerJoin(PlayerJoinEvent $event){
-        $player = $event->getPlayer();
-		
-	    if($this->getConfig()->get("joindisablefly") === true){
-		    if($player->getGamemode() === Player::CREATIVE) return;
-			    if($player->getAllowFlight() === true){
-				    $player->setFlying(false);
-				    $player->setAllowFlight(false);
-					$player->sendMessage(C::RED . $this->getConfig()->get("onjoin-flight-disabled"));
-				    return;
-			    }
-	    }
-    }
-	
-	public function BlacklistedWorldCheck($entity){
```

Comment:
Thank you for removing all these poorly camelCased methods :D

Reply:
No problem.